lib/lang/vi.js

```diff
@@ -0,0 +1,167 @@
+"use strict";
+
+function join_with_shared_prefix(a, b, joiner) {
+  let m = a;
+  let i = 0;
+
+  // HACK: This gets around "today through on Tuesday" or cases like it, which
+  // are incorrect in English.
+  if(m === "hôm nay" || m === "ngày mai")
+    m = "vào " + m;
+
+  while(i !== m.length &&
+        i !== b.length &&
+        m.charCodeAt(i) === b.charCodeAt(i))
+    ++i;
+
+  while(i && m.charCodeAt(i - 1) !== 32)
+    --i;
+
+  return a + joiner + b.slice(i);
+}
+
+function strip_prefix(period) {
+  return period.slice(0, 9) === "suốt đêm" ? period.slice(4) :
+    period.slice(0, 7) === "trong " ? period.slice(7) :
+      period;
+}
```

Comment:
Does Vietnamese actually make use of this, or can it be removed?

Reply:
I will remove it in new version